tests/conftest.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from sacred import Experiment
+from torchvision import (
+    datasets,
+    transforms,
+)
+
+
+@pytest.fixture(scope="session")
+def model() -> nn.Module:
+    class BaseModel(nn.Module):
+        def __init__(self, input_sz=32 * 32 * 3, n_classes=10):
+            super(BaseModel, self).__init__()
+            self.lin = nn.Linear(input_sz, n_classes)
+
+        def forward(self, in_tensor):
+            x = in_tensor.view(-1, 32 * 32 * 3)
+            return self.lin(x)
+
+    return BaseModel()
+
+
+@pytest.fixture(scope="session")
+def experiment(model) -> Experiment:
+    ex = Experiment("image_classification", interactive=True)
+
+    # Log hyperparameters
+    @ex.config
+    def cfg():
+        data_dir = "data/CIFAR10"  # noqa: F841
+        data_tfms = {  # noqa: F841
+            "train": transforms.Compose(
+                [
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+                ]
+            )
+        }
+        lr = 1e-2  # noqa: F841
+        bs = 128  # noqa: F841
+        n_classes = 10  # noqa: F841
+        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")  # noqa: F841
```

Comment:
Like @Raalsky mentioned, would it make sense to simplify code by hard-coding on `cpu` as CI doesn't have GPU yet.

Reply:
Yes, but I think that users could also run the tests themselves for some reasons and they might have a gpu. We could discuss it seperately